pkg/protoutils/slices.go

```diff
@@ -0,0 +1,38 @@
+package protoutils
+
+import "github.com/gogo/protobuf/proto"
+
+// SliceContains returns whether the given slice of proto objects contains the given proto object.
+func SliceContains[T proto.Message](msg T, slice []T) bool {
+	for _, elem := range slice {
+		if proto.Equal(elem, msg) {
```

Comment:
In most of the messages we have Equal defined that is faster maybe we can use it here?
```suggestion
		if elem.Equal(msg) {
```